src/averageAges.js

```diff
@@ -54,8 +75,30 @@ function calculateWomenAverageAge(people, withChildren) {
  *
  * @return {number}
  */
-function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+function calculateAverageAgeDiff(people, onlyWithSon = false) {
+  const mothers = people.map(person => person.mother);
+  const children = people.map(person => person.name);
```

Comment:
There is not a single man in people array who has `null` as name, you don't need this operation.